flink-java/src/main/java/org/apache/flink/api/java/ExecutionEnvironment.java

```diff
@@ -812,18 +854,16 @@ public JobExecutionResult execute(String jobName) throws Exception {
 
 		final Plan plan = createProgramPlan(jobName);
 		final ExecutorFactory executorFactory =
-				executorServiceLoader.getExecutorFactory(configuration);
-
-		final Executor executor = executorFactory.getExecutor(configuration);
+			executorServiceLoader.getExecutorFactory(configuration);
 
-		try (final JobClient jobClient = executor.execute(plan, configuration).get()) {
+		checkNotNull(
```

Comment:
I think a more informative message would be:

```
checkNotNull(
				executorFactory,
				"Cannot find compatible factory for the specified execution.target (=%s)",
				configuration.get(DeploymentOptions.TARGET));
```
WDYT?

Reply:
Yes correct. I just noticed that we check not null for `configuration.get(DeploymentOptions.TARGET)` so that env that doesn't set `DeploymentOptions.TARGET` will fail already.